index.d.ts

```diff
@@ -22,6 +22,8 @@
 export = Shopify;
 
 /*~ Write your module's methods and properties in this class */
+type onCallLimitsFn = (limits: Shopify.ICallLimits) => void);
+
 declare class Shopify {
```

Comment:
There is but it uses generic types while this approach provides additional type information. For example: This way on("supercalifragilisticexpialidocious") will fail while extending EventEmitter would allow it.

Reply:
Hmm, technically the `'supercalifragilisticexpialidocious'` event is valid. It will not be emitted by the lib, but the user could register and emit it manually.

```js
shopify.on('mycustomevent', listener);
shopify.emit('mycustomevent', 'because I need this');
```